### PR TITLE
frontend/mcq : Allowing task creator to (not) shuffle choices.

### PR DESCRIPTION
**Pull request by Drumor**
regarding #625 this PR allows to shuffle or not choices in MCQ

### CODE REVIEW

**Comment by anthonygego** [comment]
Didn't test it yet. Can you confirm that it allows to reload an old submission correctly ?

**Comment by Drumor** [comment]
I did test by submitting with master branch then getting back to shuffle_mcq and reload and seems to work correctly.

File: inginious/frontend/task_problems.py [hunk 1]
```diff
@@ -185,12 +185,12 @@ def show_input(self, template_helper, language, seed):
                 if entry['valid'] and limit > 0:
                     choices.append(entry)
                     limit = limit - 1
-
-        rand.shuffle(choices)
-
+        if self._shuffle:
+            rand.shuffle(choices)
+        else:
+            choices = sorted(choices, key=lambda k: k['index'])
```
Comment (anthonygego) [inline inginious/frontend/task_problems.py:191]: You short-circuit only one shuffle here, and sort an already randomly generated list. Your list of choice will be displayed in the teacher order, but the choice set will still be randomly generated according to the seed.

I don't know if this is the expected result.

Comment (Drumor) [reply inginious/frontend/task_problems.py:191]: As there is a pick of correct answer (multiple or not) just after the first shuffle, it ruins the order in any case. So i'm not sure that a course-circuit is necessary on first shuffle as the order will not be respected anyway.

Comment (anthonygego) [reply inginious/frontend/task_problems.py:191]: What would be the point of this feature then ?  The idea is to have the same rendering whatever the random seed parameters are. Currently, you will not have the same rendering if the language is different for instance.

**Comment by anthonygego** [comment]
Can this be reviewed now ?

**Comment by Drumor** [comment]
Yes! This one is good for me :)

File: inginious/common/tasks_problems.py [hunk 2]
```diff
@@ -211,6 +211,7 @@ def __init__(self, problemid, content, translations, taskfs):
         super(MultipleChoiceProblem, self).__init__(problemid, content, translations, taskfs)
         self._header = content['header'] if "header" in content else ""
         self._multiple = content.get("multiple", False)
+        self._shuffle = content.get("shuffle", False)
```
Comment (anthonygego) [inline inginious/common/tasks_problems.py:214]: The current behaviour is that answers are shuffled. Therefore, the default should be ``True``, or the name and effect of the feature inversed.

**Review by anthonygego — CHANGES_REQUESTED** [review]

File: inginious/common/tasks_problems.py (hunk 2 above)
Comment (Drumor) [reply inginious/common/tasks_problems.py]: You're right! :)

**Comment by anthonygego** [comment]
This is not the only change to make. The default value should reflect in the studio : the checkbox has to be checked if the field is not set.

**Review by anthonygego — APPROVED** [review]

**Comment by anthonygego** [comment]
LGTM

**Pull request merged**